src/common-model/functions.ts

```diff
@@ -133,27 +133,60 @@ export function getClinicalEntitySubmittedData(
 ): ClinicalInfo[] {
   const result = getClinicalObjectsFromDonor(donor, clinicalEntitySchemaName) as any[];
 
-  const clinicalRecords = result
-    .map((entity: any) =>
-      clinicalEntitySchemaName === ClinicalEntitySchemaNames.DONOR
-        ? {
+  const clinicalRecords =
+    clinicalEntitySchemaName === ClinicalEntitySchemaNames.DONOR
+      ? result.map((entity: any) => ({
+          donor_id: donor.donorId,
+          program_id: donor.programId,
+          submitter_id: donor.submitterId,
+          gender: donor.gender,
+          ...entity.clinicalInfo,
+        }))
+      : clinicalEntitySchemaName === ClinicalEntitySchemaNames.TREATMENT
+      ? result.map((treatment: any) => {
+          const clinicalInfo = treatment.clinicalInfo || {};
+          const therapy_type =
+            treatment.therapies.length === 1
+              ? { therapy_type: treatment.therapies[0].therapyType }
+              : {};
+          const therapy_info =
+            treatment.therapies.length === 1 && treatment.therapies[0].clinicalInfo;
+          return {
             donor_id: donor.donorId,
             program_id: donor.programId,
             submitter_id: donor.submitterId,
-            gender: donor.gender,
-            ...entity.clinicalInfo,
-          }
-        : {
+            treatment_id: treatment.treatmentId,
+            ...clinicalInfo,
+            ...therapy_type,
+            ...therapy_info,
+          };
+        })
+      : result
+          .filter(record => notEmpty(record.clinicalInfo))
+          .map((entity: any) => ({
             donor_id: donor.donorId,
             program_id: donor.programId,
+            submitter_id: donor.submitterId,
             ...entity.clinicalInfo,
-          },
-    )
-    .filter(notEmpty);
+          }));
 
   return clinicalRecords;
 }
 
+export const requiredEntities = (entityTypes: string[]) =>
+  // Donor Completion Stats require Sample Registration data
+  (entityTypes.includes('donor') && !entityTypes.includes('sampleRegistration')) ||
+  // Sample Registration requires Specimen data
+  (entityTypes.includes('sampleRegistration') && !entityTypes.includes('specimens'))
+    ? ['completionStats', 'sampleRegistration', 'specimens']
+    : // Clinical Therapies require Treatments
+    // hormoneTherapy + treatment do not match schema names
+    entityTypes.includes('hormoneTherapy') ||
+      entityTypes.includes('treatment') ||
+      ClinicalTherapySchemaNames.some(entity => entityTypes.includes(entity))
+    ? ['treatments', 'treatment']
+    : '';
```

Comment:
Feels like the nested ternaries are getting a bit hard to parse... you might consider moving all the comments into a single block, and this might actually be a good use for an if/else structure since it would separate the conditions.